### PR TITLE
refactor(component): refactor CheckboxOption component

### PR DESCRIPTION
**Pull request by gmukul01**
## Context

Refactor CheckboxOption

## Changes

1. Changed files of `CheckboxOption` & `CheckboxButton` component files to `typescript`.
2. Add `as` prop to `CheckboxOption` component.

## Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/components/src/checkboxButton/types.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { HTMLProps } from 'react';
+
+export type CheckboxButtonProps = HTMLProps<HTMLInputElement>;
```
Comment (dozortsev) [inline packages/components/src/checkboxButton/types.ts:3]: we tend to define types of component properties inside component file above the component code, could you please move them?

File: packages/components/src/checkboxOption/types.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+export type CheckboxOptionProps = Omit<BaseOptionProps, 'onChange'> & {
+  /**
+   * The checked state of the checkbox.
+   */
+  checked?: boolean;
+  /**
+   * Function to call when the checkbox option is clicked
+   */
+  onChange: (value: boolean) => void;
```
Comment (dozortsev) [inline packages/components/src/checkboxOption/types.ts:11]: why we need omit and then redefine `onChange` prop?

File: packages/components/src/checkboxOption/CheckboxOption.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import CheckboxButton from '../checkboxButton';
+import Option from '../common/Option';
+
+import { CheckboxOptionProps } from './types';
+
+const CheckboxOption = ({ checked, disabled, onChange, ...rest }: CheckboxOptionProps) => {
```
Comment (dozortsev) [inline packages/components/src/checkboxOption/CheckboxOption.tsx:6]: could you please also add `ref` support for this component

File: packages/components/src/common/Option/Option.tsx [hunk 4]
```diff
@@ -3,9 +3,10 @@ import { MouseEvent, forwardRef, HTMLAttributes, ReactNode, ElementType } from '
 
 import { LinkProps } from '../commonProps';
 
-interface BaseOptionProps extends Omit<HTMLAttributes<HTMLElement>, 'title'> {
+export interface BaseOptionProps extends Omit<HTMLAttributes<HTMLElement>, 'title'> {
   'aria-label'?: string;
   media?: ReactNode;
+  name?: string;
```
Comment (dozortsev) [inline packages/components/src/common/Option/Option.tsx:9]: just out of curiosity why you need `name` prop?

File: packages/components/scripts/generate-type-declarations.js [hunk 5]
```diff
@@ -24,7 +23,6 @@ const jsFiles = [
   { filename: 'Popover', folder: 'popover' },
   { filename: 'Money', folder: 'money' },
   { filename: 'RadioOption', folder: 'radioOption' },
-  { filename: 'CheckboxOption', folder: 'checkboxOption' },
```
Comment (dozortsev) [inline packages/components/scripts/generate-type-declarations.js:27]: 👏

**Review by dozortsev — COMMENTED** [review]
look great, left a few minor comments

File: packages/components/src/checkboxOption/types.ts (hunk 2 above)
Comment (gmukul01) [reply packages/components/src/checkboxOption/types.ts:11]: Actually the `onChange` prop in `BaseOptionProps ` is of `FormEventHandler<HTMLElement>`, so I had to omit that one to change it to `(value: boolean) => void`

File: packages/components/src/common/Option/Option.tsx (hunk 4 above)
Comment (gmukul01) [reply packages/components/src/common/Option/Option.tsx:9]: We are passing the name prop also in `CheckboxOption.story.tsx`
image

File: packages/components/src/checkboxButton/types.ts (hunk 1 above)
Comment (gmukul01) [reply packages/components/src/checkboxButton/types.ts:3]: Should we not keep the types in separate files for better separation of concern and better readability ?

File: packages/components/src/checkboxOption/CheckboxOption.tsx (hunk 3 above)
Comment (gmukul01) [reply packages/components/src/checkboxOption/CheckboxOption.tsx]: Done

File: packages/components/src/checkboxButton/types.ts (hunk 1 above)
Comment (dozortsev) [reply packages/components/src/checkboxButton/types.ts:3]: IMHO it's more convenient to keep Props types in component file, it's easier to scan 👀 code and understand what this component about, I think this way you get better readability

re. separation of concern, we can review this idea later, for please just follow approach that we have in all other components

File: packages/components/src/checkboxOption/types.ts (hunk 2 above)
Comment (dozortsev) [reply packages/components/src/checkboxOption/types.ts:11]: why can't we use native one (`FormEventHandler<HTMLElement>`)?

Comment (dozortsev) [reply packages/components/src/checkboxOption/types.ts:11]: why can't we use native one (`FormEventHandler<HTMLElement>`)?

File: packages/components/src/checkboxButton/types.ts (hunk 1 above)
Comment (gmukul01) [reply packages/components/src/checkboxButton/types.ts]: Perfect. Done.

File: packages/components/src/checkboxOption/types.ts (hunk 2 above)
Comment (dozortsev) [reply packages/components/src/checkboxOption/types.ts]: wait, sorry, using native interface for `onChange` in this case will lead to breaking change, which we don't want now, so please ignore my above https://github.com/transferwise/neptune-web/pull/1601#discussion_r976298284.

**Review by dozortsev — APPROVED** [review]

**Pull request merged**